design/httpproxy-conflict.md

```diff
@@ -0,0 +1,82 @@
+Contour takes a passive look to the cluster by reacting to changes after they have changed. 
+Due to this, Contour cannot block changes to resources before they are committed like an `Admission Controller` would be able to do (See alternatives: `HTTPProxyInstance`)
+
+## Detailed Design
+
+### Invalid Include
+Contour utilizes an inclusion model to allow for proper route/header delegation across namespaces and teams within a Kubernetes cluster.
+If an include does not have a matching HTTPProxy.Name or HTTPProxy.Namespace, then Contour will set an error in the Conditions of the object.
+Any requests matching the `Conditions` on the `Spec.Include` will return a `404` to the requester.
+
+If the include was valid previously and is now no longer valid, the routes will no longer be valid which the include previously enabled.
+
+The logic described above should only apply if the object has a proper delegation chain to the defined `Conditions` on the include. 
+If an object creates an `Include` to values that it does not have permission to, then it shouldn't create `404` responses.
+
+_Note: This is a use-case of an alternative model of implementing an `HTTPProxyInstance` resource which gives users a "last known good configuration" that can be reverted back to or an `Admission Controller`._
+
+### Invalid Route
+If an HTTPProxy object references a `Conditions.Prefix` that it does not have a proper delegation chain from a `root` HTTPProxy, then that resource will be marked as `orphaned`.
+Contour will set the Conditions with the orphaned status, ignore this route, but serve all other valid routes configured in the object.
+
+### Invalid or Missing Kubernetes Service
+From a `Route` in an HTTPProxy, services are referenced in a variety of places, but are primarily referenced from the `Spec.Routes` section.
+Services could be configured in error by not matching a service name or by the service not existing at all when it might of previously existed.
+
+_Note: Work has started to implementing this work in: https://github.com/projectcontour/contour/pull/3071 (Thanks @tsaarni!)_
+
+#### Single Service
+For the case where only a single service is referenced from a `Spec.Route` of an HTTPProxy, Contour will return a `503` status code and also set an error in the Conditions of the object.
+
+#### Multiple Services
+For the case where multiple services are referenced from a `Spec.Route` of an HTTPProxy, Contour will set the invalid service weight to be `zero` so that no traffic is routed to it and will also set an error in the Conditions of the object.
+
+## Alternatives Considered
+
+### Admission Controller
+An admission controller could be utilized to do these checks before the object is written to API.
+This would require a way to look at a DAG to verify status information and would require further thought into how to implement but would provide the best user experience.
```

Comment:
An Admission Controller will require the Admission controller to contain a full copy of the DAG, and run the same DAG process on admission, I think.

Reply:
Contour could also expose an endpoint that is called and asked, "hey is this valid" or something as well to keep the logic in one binary.

Reply:
Doing a hypothetical DAG run is an interesting idea, but I don't know how well it would work - I don't think that the DAG is uncoupled enough from the rest of Contour at the moment for this to fly.